core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java

```diff
@@ -179,8 +179,6 @@ public static int calculateMinShouldMatch(int optionalClauseCount, String spec)
             result = calc < 0 ? result + calc : calc;
         }
 
-        return (optionalClauseCount < result ?
-                optionalClauseCount : (result < 0 ? 0 : result));
-
+        return result < 0 ? 0 : result;
```

Comment:
do you know if there is a good reason to return 0 if we got a negative value, or could we just return 'result' directly? (if this doesn't make any tests fail, this is good enough for me)